cloudformation-telemetry/lambda_function.py

```diff
@@ -0,0 +1,148 @@
+from collections import defaultdict
+from re import T
+import boto3
+from datetime import datetime
+import os
+import json
+import requests
+import time
+from crhelper import CfnResource
+from abc import ABC, abstractmethod
+
+
+helper = CfnResource(json_logging=False, log_level='INFO', sleep_on_delete=30)
+
+@helper.create
+def create(event, context):
+    try:
+        T = telemetryFactory(event, context)
+        T.fetch_and_send_telemetry()
+    except Exception as e:
+        print(e)
+        return "Telemetry failed to sent for Create Stack"    
+    helper.Status = "SUCCESS"
+    return "Telemetry sent for Create Stack"
+
+
+@helper.update
+def update(event, context):
+    try:
+        T = telemetryFactory(event, context)
+        T.fetch_and_send_telemetry()
+    except Exception as e:
+        print(e)
+        return "Telemetry failed to sent for Update Stack"    
+    helper.Status = "SUCCESS"
+    return "Telemetry sent for Update Stack"
+
+
+@helper.delete
+def delete(event, context):
+    try:
+        T = telemetryFactory(event, context)
+        T.fetch_and_send_telemetry()
+    except Exception as e:
+        print(e)
+    helper.Status = "SUCCESS"
+
+
+def lambda_handler(event, context):
+    helper(event, context)
+
+def telemetryFactory(event, context):
+    if os.getenv['solutionName'] == 'AWSO':
+        return awsoTelemetry(event, context)
+    else:
+        return None
+    # elif os.environ['solutionName'] == 'CIS':
+    #     return cisTelemetry(event, context)
+
+# Interface
+class baseTelemetry(ABC):
+    @abstractmethod
+    def fetch_and_send_telemetry(self):
+        raise NotImplementedError
+
+    def send_telemetry(self, data):
+        # paramaterise url variable
+        url = 'https://collectors.sumologic.com/receiver/v1/http/ZaVnC4dhaV24CA_LXFO0iHFPLWH8VaEczkwtk-GZYMlTG_Dl2CPQ6YNbmKXf9K3dZQ2aAjTREC_C3TECzVQc1XN7zw5CI5lIR4O4-uYsk4bTELB1MU57AQ=='
```

Comment:
lets populate this from TELEMETRY_ENDPOINT environment variable otherwise security team will flag it.